src/middleware/default_headers.rs

```diff
@@ -0,0 +1,53 @@
+use futures::future::FutureObj;
+
+use http::{
+    header::{HeaderValue, IntoHeaderName},
+    HeaderMap, HttpTryFrom,
+};
+
+use crate::{head::Head, Middleware, Response};
+
+#[derive(Clone, Default)]
+pub struct DefaultHeaders {
+    headers: HeaderMap,
+}
+
+impl DefaultHeaders {
+    pub fn new() -> DefaultHeaders {
+        DefaultHeaders::default()
+    }
+
+    #[inline]
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::match_wild_err_arm))]
+    pub fn header<K, V>(mut self, key: K, value: V) -> Self
+    where
+        K: IntoHeaderName,
+        HeaderValue: HttpTryFrom<V>,
+    {
+        match HeaderValue::try_from(value) {
+            Ok(value) => {
+                self.headers.append(key, value);
+            }
+            Err(_) => {
+                panic!("Cannot create default header. Please check your default header values.")
```

Comment:
How about using `.expect()` on `try_from`? We don't need clippy attr then :)

Reply:
When I try to do that, I get this error:
```
the method `expect` exists but the following trait bounds were not satisfied:
      `<http::header::value::HeaderValue as http::convert::HttpTryFrom<V>>::Error : std::fmt::Debug`
```

any thoughts on how to fix that? Should I use `map_err`?

Reply:
Ok, I used `map_err`, not sure if that is the correct way to handle this or not :P